fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -58,7 +58,9 @@ class ShardedDataParallel(nn.Module):
             (default: True) Check whether the parameters trainability (`requires_grad`) has changed and update both ShardedDDP
             and OSS automatically if this is the case. If set to False, `refresh_trainable()` needs to be called anytime
             a parameter is frozen or unfrozen.
-
+        reduce_fp16 (bool):
+            cast the grads to fp16 before reducing. Not needed if the model is already fp16, but will probably improve performance
```

Comment:
Why does PyTorch convert the grads to FP32?

Reply:
It follows the model type, I discovered that recently but basically autocast walks back the graph and follows the types. It makes sense on a single gpu since the next step is the optimizer, but it's a bit unfortunate when distributed since the next step is reduction/all reduce. With ddp you can use a "fp16 compression hook" for similar reasons

Reply:
Thank you for the context! Can we add this to the PR notes or perhaps a comment here to say that this is equivalent to the fp16 compression hook? (maybe you did but I missed it?)

Reply:
done !